contracts/FeeCustody.sol

```diff
@@ -0,0 +1,410 @@
+
+    // Intermediary path asset for univ3 swaps.
+    // Empty if direct pool swap between asset and distribution asset
+    // ex: for ETH -> USDC _intermediaryPath = [] (ETH -> USDC)
+    // ex: for AAVE -> USDC _intermediaryPath = [ETH] (AAVE -> ETH -> USDC)
+    mapping(address => address[]) public intermediaryPath;
+
+    // Oracle between asset/usd pair for total
+    // reward approximation across all assets earned
+    mapping(address => address) public oracles;
+
+    // Pool fees for Univ3 swap
+    // Length 1 if direct swap between asset and distribution asset
+    // Length 2 if swap with intermediary asset
+    mapping(address => uint24[]) public poolFees;
+
+    address[100] assets;
+    // Index of empty slot in assets array
+    uint256 public lastAssetIdx;
+
+    // Events
+    event RemoveAsset(address asset);
+    event NewAsset(address asset, address[] intermediaryPath, address[] poolFees);
+    event RecoveredAsset(address asset);
+    event NewFeeDistributor(address feeDistributor);
+    event NewRBNLockerAllocation(uint256 pctAllocationForRBNLockers);
+    event NewDistributionToken(address distributionToken);
+    event NewProtocolRevenueRecipient(address protocolRevenueRecipient);
+    event NewAdmin(address admin);
+
+    /**
+     * @notice
+     * Constructor
+     * @param _pctAllocationForRBNLockers percent allocated for RBN lockers (100% = 10000)
+     * @param _distributionToken asset to distribute to RBN lockers
+     * @param _feeDistributor address of fee distributor where protocol revenue claimable
+     * @param _protocolRevenueRecipient address of multisig
+     * @param _admin admin
+     */
+    constructor(
+        uint256 _pctAllocationForRBNLockers,
+        address _distributionToken,
+        address _feeDistributor,
+        address _protocolRevenueRecipient,
+        address _admin
+    ) {
+        require(_pctAllocationForRBNLockers != 0, "!0");
+        require(_distributionToken != address(0), "!address(0)");
+        require(_feeDistributor != address(0), "!address(0)");
+        require(_protocolRevenueRecipient != address(0), "!address(0)");
+        require(_admin != address(0), "!address(0)");
+
+        pctAllocationForRBNLockers = _pctAllocationForRBNLockers;
+        distributionToken = IERC20(_distributionToken);
+        feeDistributor = IFeeDistributor(_feeDistributor);
+        protocolRevenueRecipient = _protocolRevenueRecipient;
+        admin = _admin;
+    }
+
+    // Modifier for only admin methods
+    modifier onlyAdmin {
+          require(msg.sender == admin);
+          _;
+    }
+
+    /**
+     * @notice
+     * Swaps RBN locker allocation of protocol revenu to distributionToken,
+     * sends the rest to the multisig
+     * @dev Can be called by admin
+     * @return amount of distributionToken distributed to fee distributor
+     */
+    function distributeProtocolRevenue() external onlyAdmin returns (uint256 toDistribute) {
```

Comment:
we should pass in minAmountOut for univ3 swap to avoid MEV right

Reply:
Yeap we do

Reply:
Absolutely.  If you set `minAmountOut=0` we will get sandwiched immediately and lose all your money